File: Strings/771_jewels_and_stones.py
# O(J + S)
class Solution(object):
    def numJewelsInStones(self, J, S):
        """
        :type J: str
        :type S: str
        :rtype: int
        """
        J_set = set(J)
        total = 0
        for s in S:
            if s in J_set:
                total += 1 
        return total


if __name__ == '__main__':
    J = input('J: ')
    S = input('S: ')
    print('Jewels in S: ', Solution().numJewelsInStones(J, S))

File: Strings/771_jewels_and_stones.cpp
#include <string>
#include <unordered_set>
#include <iostream>


// O(J + S)
class Solution {
public:
    int numJewelsInStones(std::string J, std::string S) {
        std::unordered_set<char> jewels;
        for (std::size_t i = 0; i < J.size(); i++)
            jewels.insert(J[i]);
        
        int sum = 0;
        for (std::size_t i = 0; i < S.size(); i++){
            if (jewels.find(S[i]) != jewels.end())
                sum++;
        }
            
        return sum;
    }
};


int main(){
    std::string J, S;
    std::cout << "J: ";
    std::cin >> J;
    std::cout << "S: ";
    std::cin >> S;

    std::cout << "Jewels in S: " << Solution().numJewelsInStones(J, S) << std::endl;
}